python/cudf/cudf/tests/test_timedelta.py

```diff
@@ -907,35 +904,34 @@ def test_timedelta_index_ops_with_scalars(data, other_scalars, dtype, op):
         expected = other_scalars // ptdi
         actual = other_scalars // gtdi
 
-    if op == "floordiv":
-        # Hand-coding pytest.xfail behaviour for certain combinations
-        if (
-            0 in ptdi.astype("int")
-            and np.timedelta64(other_scalars).item() is not None
-        ):
-            with pytest.raises(AssertionError):
-                # Related to https://github.com/rapidsai/cudf/issues/5938
-                #
-                # Division by zero for datetime or timedelta is
-                # dubiously defined in both pandas (Any // 0 -> 0 in
-                # pandas) and cuDF (undefined behaviour)
-                assert_eq(expected, actual)
-        elif (
-            (None not in ptdi)
-            and np.nan not in expected
-            and (
-                expected.astype("float64").astype("int64")
-                != expected.astype("int64")
-            ).any()
-        ):
-            with pytest.raises(AssertionError):
-                # Incorrect implementation of floordiv in cuDF:
-                # https://github.com/rapidsai/cudf/issues/12120
-                assert_eq(expected, actual)
-        else:
-            assert_eq(expected, actual)
-    else:
-        assert_eq(expected, actual)
+    # Division by zero for datetime or timedelta is
+    # dubiously defined in both pandas (Any // 0 -> 0 in
+    # pandas) and cuDF (undefined behaviour)
+    request.applymarker(
+        pytest.mark.xfail(
+            condition=(
+                op == "floordiv"
+                and 0 in ptdi.astype("int")
+                and np.timedelta64(other_scalars).item() is not None
+            ),
+            reason="Related to https://github.com/rapidsai/cudf/issues/5938",
+        )
+    )
+    request.applymarker(
+        pytest.mark.xfail(
+            condition=(
+                op == "floordiv"
+                and (None not in ptdi)
+                and np.nan not in expected
+                and (
+                    expected.astype("float64").astype("int64")
+                    != expected.astype("int64")
+                ).any()
+            ),
+            reason="https://github.com/rapidsai/cudf/issues/12393",
```

Comment:
I thought all the floor division issues were fixed, but it turns out there was special-casing in timedelta.__floordiv__; I'll handle the _fix_ separately.